File: Java/src/main/java/TopologieDto.java
import java.util.Map;

// FIXME: Dans le domain ?
public class TopologieDto {
    public Map<String, TopologieSeatDto> seats;

    public class TopologieSeatDto {
        public String booking_reference;
        public int seat_number;
        public String coach;
    }
}

File: csharp/src/TopologieDto.cs
﻿using System.Collections.Generic;

namespace KataTrainReservation
{
    public class TopologieDto
    {
        public Dictionary<string, TopologieSeatDto> seats;

        public class TopologieSeatDto
        {
            public string booking_reference;
            public int seat_number;
            public string coach;
        }
    }
}